test/common/provider.c

```diff
@@ -0,0 +1,212 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "provider.h"
+
+#include <umf/memory_provider_ops.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+static enum umf_result_t nullInitialize(void *params, void **pool) {
+    (void)params;
+    *pool = NULL;
+    return UMF_RESULT_SUCCESS;
+}
+
+static void nullFinalize(void *pool) { (void)pool; }
+
+static enum umf_result_t nullAlloc(void *provider, size_t size,
+                                   size_t alignment, void **ptr) {
+    (void)provider;
+    (void)size;
+    (void)alignment;
+    *ptr = NULL;
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t nullFree(void *provider, void *ptr, size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    return UMF_RESULT_SUCCESS;
+}
+
+static void nullGetLastError(void *provider, const char **ppMsg,
+                             int32_t *pError) {
+    (void)provider;
+    (void)ppMsg;
+    (void)pError;
+}
+
+static enum umf_result_t nullGetRecommendedPageSize(void *provider, size_t size,
+                                                    size_t *pageSize) {
+    (void)provider;
+    (void)size;
+    (void)pageSize;
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t nullGetPageSize(void *provider, void *ptr,
+
+                                         size_t *pageSize) {
+    (void)provider;
+    (void)ptr;
+    (void)pageSize;
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t nullPurgeLazy(void *provider, void *ptr, size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    return UMF_RESULT_SUCCESS;
+}
+
+static enum umf_result_t nullPurgeForce(void *provider, void *ptr,
+                                        size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    return UMF_RESULT_SUCCESS;
+}
+
+static const char *nullName(void *provider) {
+    (void)provider;
+    return "null";
+}
+
+umf_memory_provider_handle_t nullProviderCreate(void) {
+    struct umf_memory_provider_ops_t ops = {
```

Comment:
please define this globally and force using standard (umf*) API to create a null provider instead of providing this function

Reply:
Done.